dashboard/app/models/user_level.rb

```diff
@@ -130,28 +130,29 @@ def before_unsubmit
     end
   end
 
+  # `locked` is a virtual attribute because it relies on `unlocked_at` to
+  # automatically return `true` after `AUTOLOCK_PERIOD`
+  def locked
```

Comment:
it turns out that by removing the `?` at the end of this and having a matching setter, `locked` automatically becomes a "virtual attribute", meaning we can actually set it e.g. during user level creation (see user_level_test.rb line 299).

this diff is a little confusing since i moved `locked` above `locked=`, but i want to note that i didn't change the logic, i just collapsed it to a single line.

Reply:
Virtual attribute = what we were going for all along. Good find:)